msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -351,6 +252,9 @@ else if(directByteBufferClass.isInstance(buffer.base)) {
      */
     MessageBuffer(ByteBuffer bb) {
         if(bb.isDirect()) {
+            if(isUniversalBuffer) {
+                throw new IllegalStateException("Cannot create MessageBuffer from DirectBuffer");
```

Comment:
Right. This code should not be called and that's why I throw IllegalStateException.